bpf/lib/common.h

```diff
@@ -1022,6 +1022,14 @@ struct lb6_service {
 	 * slots under quarantine (otherwise zero).
 	 */
 	__u16 qcount;
+#ifdef LB_ALG_PER_SERVICE
```

Comment:
Dylan: These structures are only used by code that is itself guarded by `LB_ALG_PER_SERVICE`. So sure, the structures could always be defined regardless of `LB_ALG_PER_SERVICE`, but would that really make any meaningful difference?

Reply:
Do you have an alternative? :( The problem is there is no bit space anymore available to define which LB algorithms to use.

Reply:
The union with affinity timeout we could perhaps carve some bits from there.. perhaps like:
```
	union {
		__u32 backend_id;
                struct {
	                 	__u16 affinity_timeout;
	                 	__u8   algorithm;
	                 	__u8   pad;
                };
		__u32 l7_lb_proxy_port;
	};
```
This would mean we reduce affinity timeout to maximum 65535 seconds aka 0.75 days, which could potentially break expectations.. not sure.

Reply:
> These structures are only used by code that is itself guarded by LB_ALG_PER_SERVICE. So sure, the structures could always be defined regardless of LB_ALG_PER_SERVICE, but would that really make any meaningful difference?

We are attempting to switch out compile time variables for load time variables. I think the can make some dedicated Go loader logic to determine the size of the map dynamically based on settings, but that wouldn't be visible in C. As long all code using these fields is gated with the right load time config it can work. But it would be seriously jank.

Having a proper upgrade/downgrade path for the map would be way better long term. If we never create one this situation will only become worse over time

Reply:
Alternatively, could CO-RE handle this we we end up defining two structs once we get rid of defines for shipping pre-compiled code?

Reply:
For solving the clang-free I can take a stab at encoding the algorithm into the same __u32 as affinity_timeout. We likely need to do this via mask+bitshift instead of bitfield. I can open a GH issue with the list of follow-ups and then make it 1.17 blocker.

Reply:
(created https://github.com/cilium/cilium/issues/36112 & self-assigned)